src/main/java/de/qabel/desktop/ui/remotefs/RemoteFSController.java

```diff
@@ -43,74 +54,233 @@
-                BoxFolder folder = nav.createFolder("folder" + i);
-                nav.commit();
-                for (int j = 0; j < 10; j++) {
-                    BoxNavigation navFolder = nav.navigate(folder);
-                    BoxFolder subFolder = navFolder.createFolder("subFolder" + i + j);
-                    navFolder.commit();
-                    for (int k = 0; k < 10; k++) {
-                        BoxNavigation navSubFolder = nav.navigate(subFolder);
-                        navSubFolder.upload("File" + i + j + k, file1);
-                        navSubFolder.commit();
+    @FXML
+    protected void handleUploadFileButtonAction(ActionEvent event) {
+        FileChooser chooser = new FileChooser();
+        chooser.setTitle("Choose File");
+        List<File> list = chooser.showOpenMultipleDialog(treeTable.getScene().getWindow());
+        if (list != null) {
+            for (File file : list) {
+                try {
+                    if (selectedFolder == null || selectedFolder.getValue().name.equals(ROOT_FOLDER_NAME)) {
+                        uploadFiles(file, null);
+                    } else {
+                        uploadFiles(file, (BoxFolder) selectedFolder.getValue());
                     }
+                } catch (QblStorageException e) {
+                    e.printStackTrace();
                 }
 
             }
+        }
+        refreshTreeItem();
+    }
+
+    @FXML
+    protected void handleUploadFolderButtonAction(ActionEvent event) throws QblStorageException {
+        DirectoryChooser chooser = new DirectoryChooser();
+        chooser.setTitle("Choose Folder");
+        File directory = chooser.showDialog(treeTable.getScene().getWindow());
+        chooseUploadDirectory(directory);
+        refreshTreeItem();
+    }
+
+    @FXML
+    protected void handleCreateFolderButtonAction(ActionEvent event) {
+
+        TextInputDialog dialog = new TextInputDialog("name");
+        dialog.setHeaderText(null);
+        dialog.setTitle("Change Alias");
+        dialog.setContentText("Please specify folder name");
+        Optional<String> result = dialog.showAndWait();
+        result.ifPresent(name -> {
+            try {
+                if (selectedFolder == null || selectedFolder.getValue().name.equals(ROOT_FOLDER_NAME)) {
+                    createFolder(name, null);
+                } else {
+                    createFolder(name, (BoxFolder) selectedFolder.getValue());
+                }
+            } catch (QblStorageException e) {
+                e.printStackTrace();
+            }
+        });
+        refreshTreeItem();
+    }
+
+    @FXML
+    protected void handleDeleteButtonAction(ActionEvent event) throws QblStorageException {
+        if (selectedFolder.getParent() != null) {
+            try {
+                LazyBoxFolderTreeItem parent = (LazyBoxFolderTreeItem) selectedFolder.getParent();
+                int n = JOptionPane.showConfirmDialog(
```

Comment:
JOptionPane=swing
please use JavaFX dialogs (like the TextInputDialog in line 134